### PR TITLE
Rewrite logCurrentState to safelog

### PR DESCRIPTION
**Pull request by safreiberg**
Replaces custom yml serialization with jackson / json serialization to normal safelog, which automatically rolls as well.

## General
**Before this PR**:

**After this PR**:

==COMMIT_MSG==
==COMMIT_MSG==

**Priority**:

**Concerns / possible downsides (what feedback would you like?)**:

**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:

**Does this PR need a schema migration?**

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:

**What was existing testing like? What have you done to improve it?**:

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:

**Has the safety of all log arguments been decided correctly?**:

**Will this change significantly affect our spending on metrics or logs?**:

**How would I tell that this PR does not work in production? (monitors, etc.)**:

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

**Comment by safreiberg** [comment]
Can you confirm that client IDs are all safe to log?

File: changelog/@unreleased/pr-6419.v2.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+type: improvement
+improvement:
+  description: Rewrite logCurrentState to safelog to normal service logs rather than custom yml serialization to file.
```
Comment (jeremyk-91) [inline changelog/@unreleased/pr-6419.v2.yml:3]: Probably obvious, but might want to say that a sanitised form is safelogged, and the full version is logged at unsafe

**Review by jeremyk-91 — APPROVED** [review]
`LockClient`s should indeed be safe (for example, see the conjure API definitions).

This looks very nice - thanks for the contribution!

**Review by jeremyk-91 — APPROVED** [review]
👍  Still good

File: lock-impl/src/main/java/com/palantir/lock/logger/LockServiceStateLogger.java [hunk 2]
```diff
@@ -146,17 +105,17 @@ private Map<String, Object> generateOutstandingLocksYaml(
...
             }
         });
+        List<SimpleLockRequestsWithSameDescriptor> aggregatedRequests = new ArrayList<>();
+        outstandingRequestMap.forEach((_descriptor, builder) -> aggregatedRequests.add(builder.build()));
 
-        List<SimpleLockRequestsWithSameDescriptor> sortedOutstandingRequests =
-                sortOutstandingRequests(outstandingRequestMap.values());
-
-        return nameObjectForYamlConversion(OUTSTANDING_LOCK_REQUESTS_TITLE, sortedOutstandingRequests);
+        return sortOutstandingRequests(aggregatedRequests);
```
Comment (schlosna) [inline lock-impl/src/main/java/com/palantir/lock/logger/LockServiceStateLogger.java:118]: nit: can we remove `sortOutstandingRequests` & the multiple collection copies and just stream through:

```suggestion
        return outstandingRequestMap.values().stream()
                .map(ImmutableSimpleLockRequestsWithSameDescriptor.Builder::build)
                .sorted(Comparator.comparing(SimpleLockRequestsWithSameDescriptor::getLockRequestsCount))
                .collect(Collectors.toList());
```

File: lock-impl/src/main/java/com/palantir/lock/logger/LogState.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    @Safe
+    Map<ClientId, List<SanitizedLockRequestProgress>> getSynthesizedRequestState();
+
+    @Unsafe
+    Map<ObfuscatedLockDescriptor, LockDescriptor> getLockDescriptorMapping();
+
+    default void logTo(SafeLogger log) {
+        log.info(
+                "Lock server state",
+                SafeArg.of("oustandingLockRequests", getOutstandingRequests()),
```
Comment (schlosna) [inline lock-impl/src/main/java/com/palantir/lock/logger/LogState.java:50]: spelling nit:
```suggestion
                SafeArg.of("outstandingLockRequests", getOutstandingRequests()),
```

File: lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java [hunk 4]
```diff
@@ -1187,17 +1184,12 @@ public void logCurrentState() {
         StringBuilder logString = getGeneralLockStats();
         log.info("Current State: {}", SafeArg.of("state", logString.toString()));
 
-        try {
-            logAllHeldAndOutstandingLocks();
-        } catch (IOException e) {
-            log.error("Can't dump state to Yaml", e);
-            throw new IllegalStateException(e);
-        }
+        logAllHeldAndOutstandingLocks();
```
Comment (schlosna) [inline lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java:1187]: we might want to just combine these all into one log line statement. I'm a little concerned with how much `getGeneralLockStats()` allocates if we're logging (frequently?) when lock services are already in a potentially degraded performance state

**Comment by schlosna** [comment]
Moved comments to https://github.com/palantir/atlasdb/pull/6420

**Pull request merged**